crates/krator/src/manifest.rs

```diff
@@ -1,20 +1,30 @@
 use core::pin::Pin;
 use core::task::{Context, Poll};
 use tokio::sync::watch::{channel, Receiver, Sender};
-use tokio_stream::Stream;
+use tokio_stream::{wrappers::WatchStream, Stream};
 
 /// Wrapper for `ObjectState::Manifest` type which reflects
 /// the latest version of the object's manifest.
-#[derive(Clone)]
-pub struct Manifest<T: Clone> {
+pub struct Manifest<T: Clone + Sync + Send + std::marker::Unpin + 'static> {
     rx: Receiver<T>,
+    stream: WatchStream<T>,
 }
 
-impl<T: Clone> Manifest<T> {
+impl<T: Clone + Sync + Send + std::marker::Unpin + 'static> Clone for Manifest<T> {
```

Comment:
Nit: This is getting a little long, we should probably move it to a `where` clause in all places where it occurs